pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -20,6 +21,11 @@ in
 , version, revision ? null
 , sha256 ? null
 , src ? fetchurl { url = "mirror://hackage/${pname}-${version}.tar.gz"; inherit sha256; }
+  # The GHC to build this package with.
+  #
+  # `haskell.lib.justStaticExecutables` will add this to `disallowedRequisites`
+  # unless `needsGHCAtRuntime` is set.
+, ghc ? originalGHC
```

Comment:
Why are you threading it through in this way? It doesn't really make sense that you can freely pass `ghc = …` to `haskellPackages.mkDerivation` as that already has a predetermined GHC and no other would work. You can simply rely on `ghc` as it is already in scope, can't you?

Reply:
Hmm. I guess you're right that this is usually redundant, but I see no reason why it wouldn't work. I'll remove it for now.